### PR TITLE
Remove of static temporaries in math classes

### PR DESCRIPTION
**Pull request by code-disaster**
With this PR I want to revive an issue previously discussed in e.g. #846. It's one step towards thread safety in the math classes.

There are still some static temporaries left: Vector3.tmpMat and Matrix4.quat/quat2. The long-lived deprecated Matrix4.tmp could now be removed.

I've run a couple of tests to verify my changes, but as far as I've seen not all modified functions are covered by them. Also, I've only run the tests on desktop systems. That said, a code review and **some tests on Android/iOS** would be wise. ;) Also, feel free to try the changes with your project if you are heavily using Matrix4/Quaternion.

### CODE REVIEW

File: gdx/src/com/badlogic/gdx/math/Matrix3.java [hunk 1]
```diff
@@ -566,19 +514,19 @@ public Matrix3 transpose () {
...
+		float v01 = mata[M00] * matb[3] + mata[M01] * matb[4] + mata[M02] * matb[5];
+		float v02 = mata[M00] * matb[6] + mata[M01] * matb[7] + mata[M02] * matb[8];
+
+		float v10 = mata[M10] * matb[0] + mata[M11] * matb[1] + mata[M12] * matb[2];
+		float v11 = mata[M10] * matb[3] + mata[M11] * matb[4] + mata[M12] * matb[5];
+		float v12 = mata[M10] * matb[6] + mata[M11] * matb[7] + mata[M12] * matb[8];
+
+		float v20 = mata[M20] * matb[0] + mata[M21] * matb[1] + mata[M22] * matb[2];
+		float v21 = mata[M20] * matb[3] + mata[M21] * matb[4] + mata[M22] * matb[5];
+		float v22 = mata[M20] * matb[6] + mata[M21] * matb[7] + mata[M22] * matb[8];
```
Comment (nooone) [inline gdx/src/com/badlogic/gdx/math/Matrix3.java]: Why were the constants partly be replaced?

**Comment by xoppa** [comment]
Using varargs instead of array might cause an additional allocation, which we really want to avoid in these frequently called methods. Better move the allocation to the caller (e.g. `mul(val, new float[] { a, b, etc.})`) or add an additional signature which takes all the float as argument (e.g. `mul(float[], float, float, float, float, etc.)`)

**Comment by code-disaster** [comment]
@xoppa you are right about the allocation, I wasn't aware of that. Indeed the byte code shows newarray instructions. This of course limits the usefulness of my changes. It can be easily fixed for Matrix3, but for Matrix4 it probably needs a second native mul() function to avoid the allocation.

**Comment by xoppa** [comment]
Yeah, I'm not sure this actually solves an issue now. I guess this PR is about whether we want to complicate code for thread safety.

**Comment by MobiDevelop** [comment]
I don't want to complicate code for thread safety.

**Comment by code-disaster** [comment]
I'm ok with that. Let me close this PR then.

**Pull request closed without merging**